rust/lance-graph/src/datafusion_planner/expression.rs

```diff
@@ -476,6 +480,104 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_boolean_expr_ilike() {
+        let expr = BooleanExpression::ILike {
+            expression: ValueExpression::Property(PropertyRef {
+                variable: "p".into(),
+                property: "name".into(),
+            }),
+            pattern: "alice%".into(),
+        };
+
+        if let Expr::Like(like_expr) = to_df_boolean_expr(&expr) {
+            assert!(!like_expr.negated, "Should not be negated");
+            assert!(
+                like_expr.case_insensitive,
+                "ILIKE should be case insensitive"
+            );
+            assert_eq!(like_expr.escape_char, None, "Should have no escape char");
+            match *like_expr.expr {
+                Expr::Column(ref col_expr) => {
+                    assert_eq!(col_expr.name(), "p__name");
+                }
+                other => panic!("Expected column expression, got {:?}", other),
+            }
+            // Check pattern is a literal
+            match *like_expr.pattern {
+                Expr::Literal(ref scalar, _) => {
+                    let s = format!("{:?}", scalar);
+                    assert!(
+                        s.contains("alice%"),
+                        "Pattern should be 'alice%', got: {}",
+                        s
+                    );
+                }
+                other => panic!("Expected literal pattern, got {:?}", other),
+            }
+        } else {
+            panic!("Expected Like expression");
+        }
+    }
+
+    #[test]
+    fn test_boolean_expr_ilike_with_wildcard() {
```

Comment:
We may skip this test `test_boolean_expr_ilike_with_wildcard`, since `ilike` and `like` are very similar.